### PR TITLE
Multitargeting

### LINKED ISSUE #632: Mutitargeting.
Add ability to train against different metrics:
 - RFC
 - Fanout
 - Max diameter

### PR DESCRIPTION
**Pull request by aravij**
Fix #632 

Added ability to choose target metric to be trained against.
Currently supported:
 - Cognitive complexity
 - Fan out
 - Max methods diameter
 - Response for a class

### CODE REVIEW

File: aibolit/__main__.py [hunk 1]
```diff
@@ -120,10 +120,24 @@ def train():
...
     if not args.skip_collect_dataset:
         collect_dataset(args)
-    train_process()
+
+    metric_name_to_code = {
+        "rfc": "M9",
+        "fanout": "M10",
+        "diameter": "M6",
+        "cognitive": "M4",
+    }
```
Comment (acheshkov) [inline aibolit/__main__.py:139]: If possible, I'd add this data to aibolit/config.py

File: aibolit/model/model.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def scale_dataset(
     metrics_codes_set = [x['code'] for x in config['metrics']]
     exclude_features = set(config['patterns_exclude']).union(set(config['metrics_exclude']))
     used_codes = set(features_conf['features_order'])
-    used_codes.add('M4')
+    used_codes |= {'M4', 'M6', 'M9', 'M10'}
     not_scaled_codes = set(patterns_codes_set).union(set(metrics_codes_set)).difference(used_codes).difference(
```
Comment (lyriccoder) [inline aibolit/model/model.py:58]: can we just add a parameter, 1 parameter target. And then we will add
` used_codes.add(target)`

**Review by lyriccoder — CHANGES_REQUESTED** [review]
Set target as a parameter to scale function

File: aibolit/model/model.py (hunk 2 above)
Comment (aravij) [reply aibolit/model/model.py:59]: Done.

**Review by lyriccoder — APPROVED** [review]
ok

File: aibolit/__main__.py (hunk 1 above)
Comment (aravij) [reply aibolit/__main__.py:139]: It is possible, but this dict is specific to command line arguments of train, so I would keep it close to that method. Anyway it use different names than in config.

**Comment by lyriccoder** [comment]
@acheshkov merge it pls

**Comment by acheshkov** [comment]
@rultor merge

**Pull request merged**